Mathlib/Analysis/InnerProductSpace/Calculus.lean

```diff
@@ -245,6 +245,38 @@ theorem DifferentiableAt.norm (hf : DifferentiableAt ℝ f x) (h0 : f x ≠ 0) :
   ((contDiffAt_norm 𝕜 h0).differentiableAt le_rfl).comp x hf
 #align differentiable_at.norm DifferentiableAt.norm
 
+theorem not_differentiableAt_abs_zero : ¬ DifferentiableAt ℝ (fun x : ℝ ↦ abs x) 0 := by
+  rw [DifferentiableAt]
+  push_neg
+  intro f
+  simp [HasFDerivAt, HasFDerivAtFilter, Asymptotics.isLittleO_iff]
+  use 2⁻¹
+  norm_num
```

Comment:
this is also a bit hard to maintain as an intermediate proof step. `norm_num` calls simp so goal after this step can change if someone adds some simp lemma in the future. Can you move this step to the end of the proof somehow? Or use suffices to make it clear what the goal should be after

Reply:
This `norm_num` is meant to solve the left half of the top-level "and" in the goal. I have replaced it with `use by norm_num` to make this intention clear and the code more maintainable.